### PR TITLE
Cleanup /createRole and /deleteRole apis + frontend

### PR DESCRIPTION
**Pull request by mzparacha**
## Link to Issue
Closes: https://github.com/hicommonwealth/commonwealth/issues/4484

## Description of Changes
Removed all API calls for /createRole and /deleteRole APIs and deleted the APIs themselves

## Test Plan
- Joining a community should work correctly
- Leaving a community should work correctly

## Deployment Plan
N/A

## Other Considerations
N/A

### CODE REVIEW

File: packages/commonwealth/client/scripts/controllers/app/login.ts [hunk 1]
```diff
@@ -305,6 +305,11 @@ export async function unlinkLogin(account: AddressInfo) {
     auth: true,
     jwt: app.user.jwt,
   });
+  // remove deleted role from app.roles
+  app.roles.deleteRole({
+    address: account,
+    chain: account.chain.id,
+  })
```
Comment (mzparacha) [inline packages/commonwealth/client/scripts/controllers/app/login.ts:312]: when a user leaves a community, this code will update `app.roles` and remove the deleted role (This was a different bug I would say, not related to the cleanup work)

File: packages/commonwealth/client/scripts/views/modals/delete_address_modal.tsx [hunk 2]
```diff
@@ -45,6 +45,12 @@ export const DeleteAddressModal = (props: DeleteAddressModalAttrs) => {
         chain,
         jwt: app.user.jwt,
       });
+      // remove deleted role from app.roles
+      const foundAddressInfo = addresses.find(a => a.address === address)
+      app.roles.deleteRole({
+        address: foundAddressInfo,
+        chain: chain,
+      })
```
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/modals/delete_address_modal.tsx:53]: same reason as [here](https://github.com/hicommonwealth/commonwealth/pull/4502/files#r1263973853)

File: packages/commonwealth/client/scripts/views/modals/select_address_modal.tsx [hunk 3]
```diff
@@ -1,237 +0,0 @@
...
-          $(e.target).trigger('modalexit');
-        });
-      })
-      .catch((err: any) => {
-        setIsLoading(false);
-        notifyError(err.responseJSON.error);
-      });
-  };
-
-  const deleteRole = async (index) => {
```
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/modals/select_address_modal.tsx:73]: @jnaviask this index

File: packages/commonwealth/client/scripts/views/modals/select_address_modal.tsx [hunk 4]
```diff
@@ -1,237 +0,0 @@
...
-                    </div>
-                    <div className="role-remove">
-                      <span className="already-connected">
-                        {formatAsTitleCase(role.permission)} of{' '}
-                        {activeEntityInfo?.name}
-                      </span>
-                      <CWIcon
-                        iconName="close"
-                        iconSize="small"
-                        onClick={deleteRole.bind(this)}
```
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/modals/select_address_modal.tsx:168]: and it came from here

**Review by jnaviask — APPROVED** [review]
Seems fine to me. Please resolve merge conflicts @mzparacha

**Comment by jnaviask** [comment]
@mzparacha please fix linter / CI

**Comment by mzparacha** [comment]
will fix shortly, Thanks for the bump

**Comment by mzparacha** [comment]
Fixed @jnaviask

**Pull request merged**